beginner.md

```diff
@@ -0,0 +1,588 @@
+
+```js
+define([
+	"delite/register",
+	"delite/Widget",
+	"delite/handlebars!./BlogPost/BlogPost.html",
+    "requirejs-dplugins/css!./BlogPost/css/BlogPost.css"
+], function (register, Widget, template) {
+	return register("blog-post", [HTMLElement, Widget], {
+		baseClass: "blog-post",
+		value: "",
+		publishDate: new Date().toString(),
+		author: "",
+		template: template
+	});
+});
+
+```
+
+Note that I've added a default value for `publishDate`, to make setting the date optional; if unspecified, it will default to today's date.
+
+#####Sample usage
+So now if you change the body content of `./samples/BlogPost.html` to the following:
+
+```html
+<blog-post id="element" value="A very lazy day" publishDate="Nov 27th 2014" author="My good self"></blog-post>
+<button onclick="element.value='Now sleeping!'; event.target.disabled=true">click to change title</button>
+```
+
+And updating the template CSS `./BlogPost/css/BlogPost.css` to make it slightly more interesting too:
+
+```css
+/* style for the custom element itself */
+.blog-post {
+    display: block;
+}
+.blog-post h3 {
+    color: red;
+}
+.blog-post p.blogdetails span {
+    font-weight: bold;
+}
+/* Note this isn't used yet but will be in the next step when we discuss "delite/Container" */
+.blog-post div.blog {
+    padding-left: 20px;
+}
+```
+
+If you refresh the page you'll see it's becoming something more you'd envisage as a widget we may want to write.
+
+####delite/Container and containerNode
+Now is the time to discuss the functionality provided by [delite/Container](https://github.com/ibm-js/delite/blob/master/docs/Container.md).
+Looking at the widget we created, we need to also add arbitrary HTML to render whatever the content of our blog should be e.g. paragraph tags,
+list tags etc etc. As explained, widget properties to be displayed are really only for plain text. If you try and add any HTML to those
+properties the HTML tags will be escaped and not rendered as HTML; this is expected.
+
+As explained in the `Container` documentation, it's to be used as a base class for widgets that contain content; therefore it's also useful for our
+intentions where we want to add arbitrary HTML.
+
+#####Widget
+Let's update our widget `./BlogPost.js` to use this:
+
+```js
+define([
+	"delite/register",
+	"delite/Widget",
+	"delite/Container",
+	"delite/handlebars!./BlogPost/BlogPost.html",
+    "requirejs-dplugins/css!./BlogPost/css/BlogPost.css"
+], function (register, Widget, Container, template) {
+	return register("blog-post", [HTMLElement, Container], {
+		baseClass: "blog-post",
+		value: "",
+		publishDate: new Date().toString(),
+		author: "",
+		template: template
+	});
+});
+
+```
+
+We've extended our widget using `delite/Container`. We also only need to extend `delite/Container` because it extends `delite/Widget`.
```

Comment:
Note sure to understand this two sentences. It seems they are both saying (partially) the same thing i.e. we extend delite/Container? Maybe they can be  combined?